sys/linux/vnet.txt

```diff
@@ -1108,14 +1108,62 @@ icmp_address_reply_packet {
 	mask	int32be
 } [packed]
 
-icmp_types = ICMP_ECHOREPLY, ICMP_DEST_UNREACH, ICMP_SOURCE_QUENCH, ICMP_REDIRECT, ICMP_ECHO, ICMP_TIME_EXCEEDED, ICMP_PARAMETERPROB, ICMP_TIMESTAMP, ICMP_TIMESTAMPREPLY, ICMP_INFO_REQUEST, ICMP_INFO_REPLY, ICMP_ADDRESS, ICMP_ADDRESSREPLY
+icmp_ext_echo_packet {
+	type		const[ICMP_EXT_ECHO, int8]
+	code		const[0, int8]
+	csum		csum[parent, inet, int16be]
+	id		int16be
+	seq_num		int16be
+	icmp_ext_hdr	icmp_ext_hdr
+	iio		icmp_ext_echo_iio
+	data		array[int8]
+} [packed]
+
+icmp_ext_hdr {
+	version_reserved1	const[0, int8]
+	reserved2		const[0, int8]
+	checksum		csum[parent, inet, int16be]
+} [packed]
+
+icmp_extobj_hdr {
+	length		int16
+	class_num	int8
+	class_type	flags[icmp_ext_class_types, int8]
+} [packed]
+
+icmp_ext_echo_iio {
+	extobj_hdr	icmp_extobj_hdr
+	ident		icmp_ext_echo_iio_ident
+} [packed]
+
+icmp_ext_echo_iio_ident {
+	name		array[int8, 16]
+	ipv4_addr	int32
+	ipv6_addr	array[int8, 16]
+} [packed]
+
+icmp_ext_echo_reply_packet {
+	type		const[ICMP_EXT_ECHOREPLY, int8]
+	code		const[0, int8]
+	csum		csum[parent, inet, int16be]
+	id		int16be
+	seq_num		int16be
+	icmp_ext_hdr	icmp_ext_hdr
+	iio		icmp_ext_echo_iio
```

Comment:
Does the reply contain iio?

https://datatracker.ietf.org/doc/html/draft-bonica-intarea-eping-04#section-3